### PR TITLE
[ToggleCurrency, History] | Status: Done

### PR DESCRIPTION
**Pull request by mediano11**
**Object: [ToggleCurrencyButton]**

**Add**
    
`1. button`: add button component to toggle usd/uah currency 

**Object: [GroupHistoryPage]**

**Refactor**
    
1.`code`: refactor code

**Object: [HistoryPage]**

**Fix**
    
1.`responsive`: add responsive for history table with upd/remove buttons

### CODE REVIEW

File: src/pages/History/History.tsx [hunk 1]
```diff
@@ -18,15 +20,12 @@ import {
...
-import ConfirmationModal from '@components/ModalWindows/ConfirtmationModal/ConfirmationModal';
-import { useDeleteExpenseByGroupMutation } from '@store/Controllers/ExpensesController/ExpensesController';
-import ExpenseModal from '@components/ModalWindows/ExpenseModal/ExpenseModal';
-import { Omiter } from '@services/UsefulMethods/ObjectMethods';
+import { useAppSelector } from '@hooks/storeHooks/useAppStore';
 
 interface IColumnsHistory extends IHistoryItem {edit_remove?: string}
 
 const History: React.FC = () => {
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency);
```
Comment (tema-adidas) [inline src/pages/History/History.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/pages/GroupHistory/GroupHistory.tsx [hunk 2]
```diff
@@ -4,103 +4,29 @@ import React, {FC, useCallback, useEffect, useMemo, useState} from 'react';
...
-            <p className={classes.amount} style={{ color: info.row.original.type === 'expense' ? "#FF2D55" : "#80D667", textAlign: "left" }}>{info.row.original.type === 'expense' ? "-" : "+"}${numberWithCommas(info.getValue())}</p>,
-    }),
-]
-
+import { useAppSelector } from '@hooks/storeHooks/useAppStore';
 
 
 const History: React.FC = () => {
-
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency);
```
Comment (tema-adidas) [inline src/pages/GroupHistory/GroupHistory.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency);

File: src/components/Buttons/ToggleButton/ToggleButton.tsx [hunk 3]
```diff
@@ -1,19 +1,40 @@
...
     );
 };
 
-export default ToggleButton;
\ No newline at end of file
+export default ToggleButton;
+
+
+export const ToggleCurrencyButton: FC = () => {
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/Buttons/ToggleButton/ToggleButton.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency);

File: src/components/ChartCard/Chart.tsx [hunk 4]
```diff
@@ -27,6 +27,7 @@ type IChartProps = { setId: (Dispatch<SetStateAction<number>>), total: number; }
 )
 
 const UserExpenseChart: FC<IChartProps> = ({ categories, members, total, setId }) => {
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/ChartCard/Chart.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency);

File: src/components/ChartCard/ChartCard.tsx [hunk 5]
```diff
@@ -16,6 +17,7 @@ type IChartCardProps = { title: string, messageType?: 'user' | 'group'} & (
 
 
 const ChartCard: FC<IChartCardProps> = ({ categories, members, title, messageType = 'user' }) => {
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/ChartCard/ChartCard.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency);

File: src/components/GraphCard/Graph.tsx [hunk 6]
```diff
@@ -29,6 +29,7 @@ interface IGraphProps {
 const Graph: FC<IGraphProps> = ({data}) => {
 
     //store
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/GraphCard/Graph.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/components/GraphCard/StackedGraph.tsx [hunk 7]
```diff
@@ -36,18 +36,8 @@ type IStackedGraphProps = (
 
 const StackedGraph: FC<IStackedGraphProps> = ({ dataUsers, dataUserCategories}) => {
     //store
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/GraphCard/StackedGraph.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/components/Header/MobileHeader/MenuBurger/MenuBurger.tsx [hunk 8]
```diff
@@ -16,15 +17,19 @@ import Logo from "@assets/Header/logo.svg";
...
 interface IPropsMenuBurger {
     setMenuActive: (value: boolean) => void
     isMenuActive: boolean,
     User: IGetCurrentUserInfo
 }
 
-const MenuBurger: FC<IPropsMenuBurger> = ({setMenuActive, isMenuActive, User}) => {
+const MenuBurger: FC<IPropsMenuBurger> = ({ setMenuActive, isMenuActive, User }) => {
+    
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/Header/MobileHeader/MenuBurger/MenuBurger.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/components/OperationCard/OperationCard.tsx [hunk 9]
```diff
@@ -27,7 +24,7 @@ interface OperactionCardProps {
 }
 
 const OperationCard: FC<OperactionCardProps> = ({ operation, title, className, icon, data, isLoading, isSuccess, isError, offPreloader = false }) => {
-
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/OperationCard/OperationCard.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/components/RecentOperationsCards/RecentOperationsCards.tsx [hunk 10]
```diff
@@ -31,6 +25,7 @@ export interface IRecentOperationGroupCard {
 }
 
 export const RecentOperationDashboardCard: FC<IRecentOperationDashboardCard> = ({ item }) => {
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/RecentOperationsCards/RecentOperationsCards.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/components/RecentOperationsCards/RecentOperationsCards.tsx [hunk 11]
```diff
@@ -48,14 +43,14 @@ export const RecentOperationDashboardCard: FC<IRecentOperationDashboardCard> = (
...
-                {`${isExpense ? "-" : "+"}$${numberWithCommas(item.amount)}`}</p>
+                {`${isExpense ? "-" : "+"}${currency}${numberWithCommas(item.amount)}`}</p>
         </li>
     )
 }
 
 
 export const RecentOperationGroupCard: FC<IRecentOperationGroupCard> = ({ categoryColor, categoryTitle, time, amount, member, type }) => {
-
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/components/RecentOperationsCards/RecentOperationsCards.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/pages/Dashboard/UserAccountCard/UserAccountCard.tsx [hunk 12]
```diff
@@ -12,8 +12,7 @@ import UserAccountCardLoader from './UserAccountLoader';
 
 
 const UserAccountCard: FC = () => {
-
-    const actualTheme = useAppSelector(state => state.persistedThemeSlice.theme);
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/pages/Dashboard/UserAccountCard/UserAccountCard.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/pages/Group/GroupSpendersCard/GroupSpendersCard.tsx [hunk 13]
```diff
@@ -26,7 +26,7 @@ const GroupSpendersCard: FC<IGroupSpendersCardProps> = ({data, isLoading, isErro
         groupId: string,
         memberId: string
     }>();
-    
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```
Comment (tema-adidas) [inline src/pages/Group/GroupSpendersCard/GroupSpendersCard.tsx]: useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency)

File: src/store/UI_store/CurrencySlice/CurrencyInterfaces.ts [hunk 14]
```diff
@@ -0,0 +1,3 @@
+export interface ICurrencyState {
+    currency: '₴'|'$',
```
Comment (tema-adidas) [inline src/store/UI_store/CurrencySlice/CurrencyInterfaces.ts:2]: if we want we can add more currencies

File: src/store/UI_store/CurrencySlice/CurrencySlice.ts [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+const initialState: ICurrencyState = {
+    currency: '₴'
+}
+
+export const CurrencySlice = createSlice({
+    name: 'CurrencySlice',
+    initialState,
+    reducers: {
+        setCurrency: (initialState): void => {
+            initialState.currency = (initialState.currency === '₴') ? '$' : '₴'
```
Comment (tema-adidas) [inline src/store/UI_store/CurrencySlice/CurrencySlice.ts:15]: and here we should set like setCurrency: (initialState, action: PayloadAction<ICurrency>): void => {
            initialState.currency = (initialState.currency === '₴') ? '$' : '₴'
            
            type ICurrency = '' | '' | '' | '' | ''

**Review by tema-adidas — CHANGES_REQUESTED** [review]

File: src/store/UI_store/CurrencySlice/CurrencyInterfaces.ts (hunk 14 above)
Comment (mediano11) [reply src/store/UI_store/CurrencySlice/CurrencyInterfaces.ts:2]: we dont want, at least in 1.0

**Pull request merged**